### ISSUE TITLE
Color column based on numeric values of another column

### ISSUE THREAD

**Issue by LukasWallrich** [rstudio/gt#1103]
It would be great to be able to color one column based on the value of another - i.e. to specify a different target for `data_color`. Various SO discussions (e.g., [here](https://stackoverflow.com/questions/68673721/gt-data-color-some-columns-based-on-anothers) and [here](https://stackoverflow.com/questions/67722820/r-gt-color-a-column-by-another-columns-value)) have failed to come up with a solution, and #637 would need to be addressed to restore a (not very intuitive) workaround.

A possible desired code (and illustration for a use-case that I regularly encounter) would be the following, though something else might fit better in the gt API.

```
library(gt)

data <- data.frame(cat = LETTERS[1:3],
                   M_SD = c("1.1 (0.7)", "3.5 (0.6)", "2.0 (0.7)"),
                   M = c(1.1, 3.5, 2.0))

# DESIRED code
gt(data) %>% 
  cols_hide(3) %>%
  data_color(SOURCE_columns = 3, TARGET_columns = 2, 
    colors = scales::col_numeric(palette = c("red", "green"), domain = c(1, 4)))
```

**Comment by LukasWallrich** [comment]
For my own use, I now adapted the `data_color` function accordingly. I am sure there are edge cases when this doesn't work, and there is probably a more consistent way to implement this into gt - so I mostly wanted to add this here as a workaround for anyone else who might come across this issue: https://gist.github.com/LukasWallrich/4a91e1831db76eb66d4c583ee371ba38

**Comment by rich-iannone** [comment]
@LukasWallrich I've now addressed this in #1147 . Try it out and let me know what you think!

**Comment by rich-iannone** [comment]
Love to hear it. I did consider adding `target_rows` but deferred that work because I wanted to think a bit on how summary rows could be better targeted. But I *will* revisit this function to add that feature.
